apisix/ssl.lua

```diff
@@ -252,9 +253,14 @@ function _M.check_ssl_conf(in_dp, conf)
         end
     end
 
-    local ok, err = validate(conf.cert, conf.key)
-    if not ok then
-        return nil, err
+    -- if the certificate uses a secret reference, we only verify it when using it
+    if not secret.check_secret_uri(conf.cert) and
+        not secret.check_secret_uri(conf.key) then
+
```

Comment:
what about only one use a secret? should we check this?

Reply:
Let me adjust it. If any of the certificate and key use secret, no verification will be done.